app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, %i(edit update)
+
+  def new; end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t(".not_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      flash[:success] = t(".reset_password")
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].dowcase
```

Comment:
`.dowcase` ủa, nó chạy dc hà em? (khoc)

chị thay `downcase` thành `downcase!` được ko

Reply:
dạ không thay được   , nếu dowcase! sẽ trả về nill nếu email đó toàn chữ thường chị